packages/grafana-ui/src/components/VizLegend/SeriesIcon.tsx

```diff
@@ -1,20 +1,44 @@
 import React, { CSSProperties } from 'react';
+import { useTheme2 } from '../../themes';
+import { fieldColorModeRegistry } from '@grafana/data';
 
 export interface Props extends React.HTMLAttributes<HTMLDivElement> {
-  color: string;
+  color?: string;
+  gradient?: string;
 }
 
-export const SeriesIcon = React.forwardRef<HTMLDivElement, Props>(({ color, className, ...restProps }, ref) => {
-  const styles: CSSProperties = {
-    backgroundColor: color,
-    width: '14px',
-    height: '4px',
-    borderRadius: '1px',
-    display: 'inline-block',
-    marginRight: '8px',
-  };
+export const SeriesIcon = React.forwardRef<HTMLDivElement, Props>(
```

Comment:
@dprokop or @torkelo can you check the changes I did to the GrafanaUI components whether it makes sense?

I added optional gradient property here so you can show a gradient as the small color line next to a legend item. The usecase is that in nodeGraph you can color the nodes with a gradient based on some stat. This gives you some clue in the legend of what that stat is and what colors are associated with it.